apps/user/services/authentication.go

```diff
@@ -89,6 +93,102 @@ func (service AuthenticationService) BasicLogin(c *gin.Context) (
 	return userResponse, http.StatusAccepted, nil
 }
 
+// Identify ...
+func (service AuthenticationService) Identify(c *gin.Context) (
+	status int,
+	err error) {
+
+	var identifyData userSerializers.Identify
+	err = c.BindJSON(&identifyData)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+	gormDB := service.DB
+	var userData userModels.User
+	err = gormDB.Model(&userModels.User{}).Where("email = ?", identifyData.Email).Scan(&userData).Error
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return http.StatusBadRequest, fmt.Errorf("We couldn't find a iReflect account associated with %s", identifyData.Email)
+		}
+		return http.StatusInternalServerError, err
+	}
+	// when we don't need OTP to the email.
+	if !identifyData.SendOTP {
+		return http.StatusOK, nil
+	}
+	var otp userModels.OTP
+	err = gormDB.Model(&userModels.OTP{}).Where("user_id = ?", userData.ID).Scan(&otp).Error
+	if err != nil && err != gorm.ErrRecordNotFound {
+		return http.StatusInternalServerError, err
+	}
+	if err != gorm.ErrRecordNotFound {
+		if otp.ExpiryAt.Unix()-constants.OTPExpiryTime+constants.OTPReCreationTime > time.Now().Unix() {
+			return http.StatusBadRequest, errors.New("You just generated a OTP. Please try again after sometime")
+		}
+		err = gormDB.Delete(&otp).Error
+		if err != nil {
+			return http.StatusInternalServerError, err
+		}
+	}
+	newOTP := userModels.OTP{
+		UserID: userData.ID,
+	}
+	err = gormDB.Create(&newOTP).Error
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+	err = sendOTPAtEmail(identifyData.Email, newOTP.Code, userData.FirstName, userData.LastName)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+	return http.StatusOK, nil
+}
+
+func sendOTPAtEmail(email string, code string, firstName string, lastName string) error {
+	message, _ := parseTemplate("apps/user/views/mail.html", map[string]interface{}{"firstName": firstName, "lastName": lastName, "code": code})
+	subject := "Subject: " + "One Time Password" + "\n"
```

Comment:
Pick all the constants from the constants file.

Reply:
done.